src/memory/title_sequence_manager.rs

```diff
@@ -29,104 +47,352 @@ impl MemoryManagerUpdate for TitleSequenceManagerData {
         ctx: &StateContext,
         manager: &mut UnityMemoryManager,
     ) -> Result<(), Error> {
-        match self.update_title_menu(ctx, manager) {
-            Ok(_) => Ok(()),
-            Err(error) => Err(error),
+        if let Some(class) = manager.class {
```

Comment:
Should maybe take some time to review this pattern here. Yes, we need to check that each of these `Option` values actually exist, but since they are always used together and in specific ways, maybe we can make a transport type here so there's only one param passed into the update functions below.

This could then be used to write a method on said type to reduce this repeated pattern a bit:
```rust
      if class
          .follow_fields::<u8>(
              singleton,
              process,
              module,
              &["titleScreen", "continueButton", "selected"],
          )
          .ok()
          == Some(1)
```

Reply:
realistically we are just checking they're valid, we can probably pass the `ctx` through and use them directly from that.

Reply:
will be handled in https://github.com/Eein/sea-of-stars-tas/issues/35